test/framework/src/main/java/org/elasticsearch/index/replication/ESIndexLevelReplicationTestCase.java

```diff
@@ -177,9 +181,21 @@ protected DiscoveryNode getDiscoveryNode(String id) {
                 }
             });
 
+        private final RetentionLeaseSyncer retentionLeaseSyncer = new RetentionLeaseSyncer() {
+            @Override
+            public void sync(ShardId shardId, RetentionLeases retentionLeases, ActionListener<ReplicationResponse> listener) {
+                syncRetentionLeases(shardId, retentionLeases, listener);
+            }
+
+            @Override
+            public void backgroundSync(ShardId shardId, RetentionLeases retentionLeases) {
+                sync(shardId, retentionLeases, ActionListener.wrap(r -> {}, e -> fail("fail to sync retention leases [" + e + "]")));
```

Comment:
I slightly prefer `throw AssertionError("failed to sync retention leases", e);` rather than putting the inner stack trace in the message like this.

Reply:
done